6.19.0-6 Build Failure

Seeing the following in the build file when using current kernel 6.19 on Ubuntu Resolute:

```
Building module(s)
# command: 'make' -j 24 KVER=6.19.0-6-generic src=/usr/src/rtl88x2bu-5.8.7.1
make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/6.19.0-6-generic/build M=/var/lib/dkms/rtl88x2bu/5.8.7.1/build  modules
make[1]: Entering directory '/usr/src/linux-headers-6.19.0-6-generic'
make[2]: Entering directory '/var/lib/dkms/rtl88x2bu/5.8.7.1/build'
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: x86_64-linux-gnu-gcc (Ubuntu 15.2.0-13ubuntu3) 15.2.0
  You are using:           gcc (Ubuntu 15.2.0-12ubuntu1) 15.2.0
  CC [M]  core/rtw_cmd.o
...
  CC [M]  core/rtw_mp.o
  LD [M]  88x2bu.o
88x2bu.o: error: objtool: mp_xmit_packet_thread+0x19d: thread_exit() missing __noreturn in .c/.h or NORETURN() in noreturns.h
make[4]: *** [/usr/src/linux-headers-6.19.0-6-generic/scripts/Makefile.build:505: 88x2bu.o] Error 1
make[4]: *** Deleting file '88x2bu.o'
make[3]: *** [/usr/src/linux-headers-6.19.0-6-generic/Makefile:2060: .] Error 2
make[2]: *** [/usr/src/linux-headers-6.19.0-6-generic/Makefile:248: __sub-make] Error 2
make[2]: Leaving directory '/var/lib/dkms/rtl88x2bu/5.8.7.1/build'
make[1]: *** [Makefile:248: __sub-make] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-6.19.0-6-generic'
make: *** [Makefile:2395: modules] Error 2
```

Doing testing with the built-in rtw88_ driver in the meantime to see if they've fixed whatever issue always caused it to start failing after a couple of days.

--- a/include/osdep_service.h
+++ b/include/osdep_service.h
@@ -421,7 +421,7 @@ static __inline void thread_enter(char *name)
 	printf("%s", "RTKTHREAD_enter");
 #endif
 }
-void thread_exit(_completion *comp);
+void __noreturn thread_exit(_completion *comp);
 void _rtw_init_completion(_completion *comp);
 void _rtw_wait_for_comp_timeout(_completion *comp);
 void _rtw_wait_for_comp(_completion *comp);

--- a/os_dep/osdep_service.c
+++ b/os_dep/osdep_service.c
@@ -1296,7 +1296,7 @@ u32 _rtw_down_sema(_sema *sema)
 #endif
 }
 
-inline void thread_exit(_completion *comp)
+inline void __noreturn thread_exit(_completion *comp)
 {
 #ifdef PLATFORM_LINUX
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 17, 0))

